js/js.translator/src/org/jetbrains/kotlin/js/translate/declaration/ClassTranslator.kt

```diff
@@ -83,11 +89,30 @@ class ClassTranslator private constructor(
         translatePropertiesAsConstructorParameters(nonConstructorContext)
         val bodyVisitor = DeclarationBodyVisitor(descriptor, nonConstructorContext, enumInitFunction)
         bodyVisitor.traverseContainer(classDeclaration, nonConstructorContext)
+
+        val companionDescriptor = descriptor.companionObjectDescriptor
+
+        // add non-declared (therefore, not traversed) synthetic companion object
+        if (companionDescriptor is SyntheticClassOrObjectDescriptor) {
+            bodyVisitor.generateClassOrObject(companionDescriptor.syntheticDeclaration, nonConstructorContext, true)
+        }
+
+        // synthetic nested classes
+        descriptor
```

Comment:
what for do we to generate any other synthetic nested class?

Reply:
In JVM, they are used to generate annotation interfaces synthetic implementation. In JS, they're currently not used (because annotations translated to usual class), but I've made this call to extension for consistency, because it's possible for any extension to declare such classes.